src/main/java/org/team340/robot/subsystems/Intake.java

```diff
@@ -0,0 +1,58 @@
+package org.team340.robot.subsystems;
+
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.hardware.TalonFX;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj2.command.Command;
+import java.util.function.DoubleSupplier;
+import org.team340.lib.util.command.GRRSubsystem;
+import org.team340.lib.util.vendors.PhoenixUtil;
+import org.team340.robot.Constants.RobotMap;
+
+public class Intake extends GRRSubsystem {
+
+    final double kIntakingSpeed = 0.0;
+    final double kBarfingSpeed = 0.0;
+
+    final TalonFX intakeMotor;
+    final DigitalInput beamBreak;
+
+    public Intake() {
+        intakeMotor = new TalonFX(RobotMap.kIntakeMotor);
+
+        TalonFXConfiguration config = new TalonFXConfiguration();
+
+        PhoenixUtil.run("Clear Intake Motor Sticky Faults", intakeMotor, () -> intakeMotor.clearStickyFaults());
+        PhoenixUtil.run("Apply Intake Motor TalonFXConfiguration", intakeMotor, () ->
+            intakeMotor.getConfigurator().apply(config)
+        );
+
+        beamBreak = new DigitalInput(RobotMap.kIntakeBeamBreak);
+    }
+
+    public void setTargetSpeed(double speed) {
+        intakeMotor.set(speed);
+    }
+
+    public boolean hasPiece() {
```

Comment:
Should discuss the naming of this, the beam break is mounted somewhat outside of the hopper. "seesCoral" may be more appropriate